core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts

```diff
@@ -67,6 +77,37 @@ export class EmaContentletToolsComponent {
         }
     ];
 
+    ngOnDestroy(): void {
+        // Remove all the appended elements
+        document.querySelectorAll('[data-dot-drag-item]').forEach((el) => {
+            el.remove();
+        });
+    }
+
+    dragStart(event: DragEvent, payload: ActionPayload): void {
+        const element =
+            document.querySelector(`[data-dot-drag-item="${payload.contentlet.contentType}"]`) ??
+            document.createElement('div');
+
+        element.innerHTML = payload.contentlet.contentType;
+        element.setAttribute(
+            'style',
+            'position: absolute;top: -1000px;left: -1000px;padding: 1rem;width: fit-content;background-color: white;border: 1px solid #d1d4db;border-radius: 0.5rem;'
+        );
+
+        element.setAttribute('data-dot-drag-item', payload.contentlet.contentType);
+
+        document.body.appendChild(element);
```

Comment:
Can we use this? https://angular.io/api/core/ViewContainerRef

What I don't know is how to append it to the body, maybe @KevinDavilaDotCMS knows?

Reply:
We can, but to modify DOM is recomended [Renderer2](https://angular.io/api/core/Renderer2)

Reply:
We can create a component to render a Card, but there's a catch, we need to make a request to get the icon. I'm not sure if we want to create component for that or we stick to create an element with `Renderer2` which I have not tried yet

Reply:
Found a better approach, didn't need any of that